src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -618,24 +648,28 @@ def _getBlockDeviceMapping(cls, instanceType, rootVolSize=50):
 
     @classmethod
     def _getNodesInCluster(cls, ctx, clusterName, nodeType=None, preemptable=False, both=False):
-        for attempt in retry(predicate=AWSProvisioner._throttlePredicate):
+        for attempt in retry(predicate=awsRetryPredicate):
             with attempt:
                 pendingInstances = ctx.ec2.get_only_instances(filters={'instance.group-name': clusterName,
                                                                        'instance-state-name': 'pending'})
-        for attempt in retry(predicate=AWSProvisioner._throttlePredicate):
+        for attempt in retry(predicate=awsRetryPredicate):
             with attempt:
                 runningInstances = ctx.ec2.get_only_instances(filters={'instance.group-name': clusterName,
                                                                        'instance-state-name': 'running'})
         if nodeType:
             pendingInstances = [instance for instance in pendingInstances if instance.instance_type == nodeType]
             runningInstances = [instance for instance in runningInstances if instance.instance_type == nodeType]
-        instances = set(pendingInstances)
+        # combine lists of instances by checking id for uniqueness to prevent a lucky
```

Comment:
How about making a single call to get_only_instances without the instance-state-name filter, then separate the lists by state?

Reply:
That sounds much cleaner!